server/controller/apis/goalController.js

```diff
@@ -203,8 +201,9 @@ exports.saveGoal = async ( req, res, redirect ) =>{
         }
         else if ( !req.files || Object.keys( req.files ).length === 0 ) {
             // no files uploaded
-            this.saveGoalImage( req, res, goal.id, 'peak.svg' );
-            
+            if (!goal.goalImage){
```

Comment:
Why is this a nested if? (and what would happen if goal.goalImage exists?)
Could this just be part of the else if it is nested in?

Also I have a feeling the if statement above:
if ( req.body.goalModified && req.body.goalModified != "false" && !req.files ) {
might have been intended for the same purpose? Do we need both?

Reply:
if ( req.body.goalModified && req.body.goalModified != "false" && !req.files ) {

checks for a field (goalModified) sent from client/agora/views/dashboard/partials/goal/edit-goal.ejs, but goalModified doesn't exist in the back end. I think that's why this if/else has a logic error on back-end API calls.

Testing in the front end also reveals that the image gets deleted if the goal is updated without providing a new image. I've saved the log statements from the console and am setting a Calendly meeting tomorrow to discuss a best approach.